### DISCUSSION TITLE
How do I create a package for a CLI-only tool correctly?

### DISCUSSION THREAD

**Discussion by ero-qt** [microsoft/winget-pkgs#195085] [Q&A] [answered]
Hello,

I'm in contact with the [Extism](https://github.com/Extism) team and would like to upload the [Extism CLI](https://github.com/extism/cli) to winget. The download at https://github.com/extism/cli/releases/latest is of a `.zip` archive containing the CLI (`extism.exe`), the `LICENSE`, and a `README.md` file.

How do I correctly create a package that adds `extism` as a command to the PATH? Are the other two files an issue?

**Comment by mdanish-kh** [comment, marked as answer]
@just-ero You can use [`wingetcreate`](https://github.com/microsoft/winget-create/releases/tag/v1.6.5.0) (can get it through `winget install wingetcreate` as well) and then do

```
wingetcreate new https://github.com/extism/cli/releases/download/v1.6.1/extism-v1.6.1-windows-amd64.zip
```

The tool will take you through a series of questions to help you author a package. For reference, I can give you some values that you can use to fill up while going through the command

PackageIdentifier -> `Extism.CLI`
PackageVersion -> `1.6.1`
Is this a portable package? -> `yes`
What is the command alias of the portable package? -> `extism`
